src/ReturnBookUI.java

```diff
@@ -1,87 +1,81 @@
 import java.util.Scanner;
 
-
 public class ReturnBookUI {
 
-	public static enum UI_STATE { INITIALISED, READY, INSPECTING, COMPLETED };
+    public static enum UI_STATE {INITIALISED, READY, INSPECTING, COMPLETED};
+    
+    private ReturnBookControl control;
+    private Scanner input;
+    private UI_STATE state;
+
+    
+    public ReturnBookUI(ReturnBookControl control) {
+        this.control = control;
+        input = new Scanner(System.in);
+        state = UI_STATE.INITIALISED;
+        control.setUI(this);
+    }
+
+    
+    public void run() {
+        output("Return Book Use Case UI\n");
+        while (true) {
+            switch (state) {
+                case INITIALISED: {
+                    break;
+                }
+                case READY: {
+                    String bookStr = input("Scan Book (<enter> completes): ");
+                    if (bookStr.length() == 0) {
+                        control.scanningComplete();
+                    } 
+                    else {
+                        try {
+                            int bookId = Integer.valueOf(bookStr).intValue();
+                            control.bookScanned(bookId);
+                        } 
+                        catch (NumberFormatException e) {
+                            output("Invalid bookId");
+                        }
+                    }
+                    break;
+                }
+                case INSPECTING: {
+                    String ans = input("Is book damaged? (Y/N): ");
+                    boolean isDamaged = false;
+                    if (ans.toUpperCase().equals("Y")) {
+                        isDamaged = true;
+                    }
+                    control.dischargeLoan(isDamaged);
+                }
 
-	private ReturnBookControl control;
-	private Scanner input;
-	private UI_STATE state;
+                case COMPLETED: {
+                    output("Return processing complete");
+                    return;
+                }
 
-	
-	public ReturnBookUI(ReturnBookControl control) {
-		this.control = control;
-		input = new Scanner(System.in);
-		state = UI_STATE.INITIALISED;
-		control.setUI(this);
-	}
+                default: {
+                    output("Unhandled state");
+                    throw new RuntimeException("ReturnBookUI : unhandled state :" + state);
+                }
+            }
+        }
+    }
 
+    private String input(String prompt) {
```

Comment:
D13 - Change input() -> getInput() (differentiate from the input variable used in the same class file)